Extension/src/LanguageServer/extension.ts

```diff
@@ -770,64 +770,64 @@ async function suggestInsidersChannel(): Promise<void> {
     }
 }
 
-function applyUpdate(buildInfo: BuildInfo): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-        tmp.file({postfix: '.vsix'}, async (err, vsixPath, fd, cleanupCallback) => {
-            if (err) {
-                reject(new Error('Failed to create vsix file'));
-                return;
-            }
+async function applyUpdate(buildInfo: BuildInfo): Promise<void> {
+    const tempVSIX: tmp.FileResult = await util.CreateTempFileWithPostfix('.vsix');
```

Comment:
I believe you need a try/catch around this.  The helper function always returns a file when it succeeds, but throws an error when it fails (`reject` -> `throw`)